ext/src/content/content.js

```diff
@@ -417,42 +221,242 @@ function init() {
       });
       const newCommentObserver = new MutationObserver(() => {
         commentObserver.disconnect();
-        setupCommentObserver(commentObserver);
+        resetGithubCommentObserver(commentObserver);
       });
       const pageObserver = new MutationObserver(() => {
         if (window.location.pathname !== currentPath) {
           currentPath = window.location.pathname;
           initGitHubButton();
 
           commentObserver.disconnect();
-          setupCommentObserver(commentObserver);
-          const content = document.querySelector(".js-disscussion");
-          if (content) {
-            newCommentObserver.disconnect();
-            newCommentObserver.observe(content, { childList: true });
-          }
+          resetGithubCommentObserver(commentObserver);
+          resetGithubNewCommentObserver(newCommentObserver);
         }
       });
       pageObserver.observe(document.querySelector("body"), {
         childList: true
       });
-      const jsDiscussion = document.querySelector(".js-discussion");
-      if (jsDiscussion) {
-        newCommentObserver.observe(jsDiscussion, {
-          childList: true
-        });
-      }
-      setupCommentObserver(commentObserver);
+      resetGithubCommentObserver(commentObserver);
+      resetGithubNewCommentObserver(newCommentObserver);
       isGithubListenerAdded = true;
     }
     initGitHubButton();
   }
 
+  // Pasting events are used to force stackoverflows WMD to save state.
+  // TODO: This solution is buggy and does not always work - more discussion needed on solution
+  function stackOverflowFormat(event, inputEl) {
+    event.preventDefault();
+    const codeBlocks = getCodeBlocksStackOverflow(inputEl);
+
+    let markdownText = inputEl.value;
+    const formattedText = [];
+    if (codeBlocks.length) {
+      /*
+       * TODO: Add support for language-all: <!-- language-all: lang-* -->
+       * Once this support is added, we can format inline code snippets (i.e. `const foo = 'bar';`).
+       * https://stackoverflow.com/editing-help#syntax-highlighting
+       */
+
+      codeBlocks.forEach(block => {
+        const arrToAdd = [];
+        if (block.content.length) {
+          arrToAdd.push(
+            prettier.format(block.content.join("\n"), {
+              parser: PARSERS_LANG_MAP[block.parser],
+              plugins: prettierPlugins
+            })
+          );
+        }
+        if (block.codeBlock) {
+          arrToAdd.unshift(block.firstLine);
+          if (block.lastLine) {
+            arrToAdd.push(block.lastLine);
+          }
+        }
+        formattedText.push(...arrToAdd);
+      });
+    } else {
+      markdownText = prettier.format(inputEl.value, {
```

Comment:
The previous iteration always ran Prettier using the markdown editor on the entire value after running Prettier on the individual code blocks. I think it would still be nice to do this, because it formats outside the code blocks as well (since the markdown parser doesn't format the StackOverflow-specific code block syntax).